src/constants/z-indices.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * This file cannot be converted to TypeScript because it's referenced in the
+ * Tailwind config.
+ */
+
+/**
+ * mapping of z-index names to z-indices; Ensure that these values are also
+ * written in the `tailwind.safelist.txt` file in the project root.
+ *
+ * @type {Record<string, string|number>}
+ */
+const Z_INDICES = {
+  auto: 'auto',
+  10: 10,
+  20: 20,
+  30: 30,
+  40: 40,
+  50: 50,
+  // Named indices
+  popover: 50,
+}
+
+/**
+ * Check whether the given z-index is valid and is configured in Tailwind.
+ *
+ * @param value {string|number} the provided z-index to validate
+ * @return {boolean} whether the z-index is valid and configured in Tailwind
+ */
+const zIndexValidator = (value) =>
+  Object.keys(Z_INDICES).includes(value.toString())
+
+/**
+ * This module is consumed by Nuxt and Tailwind config, so it needs to use CJS modules.
+ */
+module.exports = {
```

Comment:
The same comment applies here as in the Screens PR: how could we move away from CommonJS modules?

Reply:
@obulat I think the limitation is in Tailwind. [Tailwind only supports CJS modules for the configuration file](https://tailwindcss.com/docs/configuration). The only alternatives I can think of are to add a build step for the Tailwind configuration that would take a `tailwind.config.ts` or `.mjs` and output the `tailwind.config.js` that Tailwind would actually consume. This adds a bit of complexity that we can just ignore by using CJS for these few specific files that are likely not touched that often.